lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3711,6 +3712,25 @@ LogicalResult FIRRTLLowering::visitStmt(VerifCoverIntrinsicOp op) {
   return success();
 }
 
+LogicalResult FIRRTLLowering::visitExpr(HasBeenResetIntrinsicOp op) {
+  auto clock = getLoweredValue(op.getClock());
+  auto reset = getLoweredValue(op.getReset());
+  if (!clock || !reset)
+    return failure();
+  auto resetType = op.getReset().getType();
+  auto uintResetType = dyn_cast<UIntType>(resetType);
+  auto isSync = uintResetType && uintResetType.getWidth() == 1;
+  auto isAsync = isa<AsyncResetType>(resetType);
+  if (!isAsync && !isSync) {
+    auto d = op.emitError("uninferred reset passed to 'has_been_reset'; "
+                          "requires sync or async reset");
+    d.attachNote() << "reset is of type " << resetType
+                   << ", should be '!firrtl.uint<1>' or '!firrtl.asyncreset'";
+    return failure();
```

Comment:
Does this actually matter? Can this not just let the reset be inferred?

Reply:
The tests below suggest that inferred reset is okay, and the fact that this is in file `LowerToHW.cpp` is leading me to believe that this error will only occur if the reset isn't properly inferred, which makes sense to me.

Reply:
Yep. When we hit `LowerToHW`, the expectation is that `InferResets` has run. If we still encounter `Reset`, the pass should complain. We don't do this in one centralized location yet, and I'm pretty sure we're not rejecting the `Reset` type consequently everywhere it can come up (`Reg` for example); but at least the new code can reject it 😁